internal/standby/standby.go

```diff
@@ -12,17 +12,27 @@ import (
 	mqtt "github.com/eclipse/paho.mqtt.golang"
 )
 
+type ServiceMode string
+
+var (
+	StandbyMode = ServiceMode("standby")
+	CommandMode = ServiceMode("command")
+)
+
 type Service struct {
-	logger     *slog.Logger
-	cfg        config.Config
-	MQTTClient mqtt.Client
+	logger                *slog.Logger
+	cfg                   config.Config
+	MQTTClient            mqtt.Client
```

Comment:
What are the reasons to make these fields public? Normally we don't want to expose such dependencies in order to maintain better separation of concerns.

Reply:
I made them public so I could access them in the test. I've instead made them private and included the test in the `standby` package

Reply:
Thanks @esimonov . I've been assuming I couldn't test this with a mock, as in order to set the `latestCommandReceived` timestamp, the standby service needs to receive an MQTT message to a particular topic and execute the handler that has been defined for it. A mock would need to invoke the handler function directly which presumably also should not be part of the public API. 

But moving the timestamp to a separate storage mechanism, as you suggest below, would help with that right? As I could then provide a mock storage dependency and set the timestamp independently of the `standby` package?

Reply:
Absolutely. You can make your mocked storage return any values of `latestCommandReceived`, and assert that the `Service` is correctly interpreting all of them. The fact that MQTT client is being called via `subscribeToTopic` can make mocking a bit less straightforward, but I don't foresee any reasons why it can't be done.

Reply:
I've tried providing an interface and wrapper for the MQTT client and the tokens that it returns, to enable mocking, but this got much more complex than I anticipated so it's [over in a different PR](https://github.com/EvergenEnergy/remote-standby/pull/10) for now and still WIP.

Reply:
Thanks for your very helpful suggestions on this PR @esimonov 🙌   I think I've addressed all of them except for mocking the MQTT client, which I tried to do in [this other branch](https://github.com/EvergenEnergy/remote-standby/pull/10), but it turned out to be quite complex with a lot of additional interfaces and wrappers, and even then didn't allow me to unit test the inner callback functions. After discussion with @jlordiales we decided it probably wasn't worth the amount of complexity it would introduce, so I have stuck with integration tests.